l6-SC.php

```diff
@@ -0,0 +1,133 @@
+<!DOCTYPE html>
+<html>
+<head>
+	<title>Thanh Nam</title>
+</head>
+<body>
+	<h1>Tính chu vi và diện tích</h1>
+	<form action="" method="GET">
+		<label>Điền vào chiều dài (hoặc chiều cao, bán kính):  </label>
+		<input type="text" name="number1"><br><br>
+
+		<label>Điền vào chiều rộng (hoặc đáy):</label>
+		<input type="text" name="number2"><br><br>
+
+
+		<input type="radio" id="tg" name="hinh" value="tamGiac"> Tam giác
+		<input type="radio" id="hcn" name="hinh" value="hinhChuNhat"> Hình chữ nhật
+		<input type="radio" id="ht" name="hinh" value="hinhTron"> Hình tròn
+		<input type="radio" id="hv" name="hinh" value="hinhVuong"> Hình vuông
+
+		<br><br>
+		<input type="submit" value="OK">
+	</form>
+
+	<?php
+
+		class Hinh 
+		{
+			public $number1;
+			public $number2;
+
+			public function __construct($number1, $number2) {
+				$this->number1 = $number1;
+				$this->number2 = $number2;
+			}
+
+			function chuVi() {}
+
+			function dienTich() {}
+		}
+
+		class HinhVuong extends Hinh
+		{
+
+			public function chuVi() {
+				$n1 = $this->number1;
+				$chuVi = $n1 * 4;
+
+				echo '<div class="hv">Chu vi hình vuông là: ' . $chuVi . '</div><br>';
+				
+			}
+
+			public function dienTich() {
+				$n1 = $this->number1;				
+				$dienTich = $n1 * $n1 ;
+
+				echo '<div class="hv">Diện tích hình vuông là: ' . $dienTich . '</div><br>';
+			}
+		}
+
+		class HinhChuNhat extends Hinh 
+		{
+
+			public function chuVi() {
+				$n1 = $this->number1;
+				$n2 = $this->number2;
+				$chuVi = ($n1 + $n2) * 2;
+
+				echo '<div class="hcn">Chu vi hình chữ nhật là: ' . $chuVi . '</div><br>';
+			}
+
+		 	public function dienTich() {
+		 		$n1 = $this->number1;
+				$n2 = $this->number2;
+				$dienTich = $n1 * $n2 ;
+
+				echo '<div class="hcn">Diện tích hình chữ nhật là: ' . $dienTich . '</div><br>';
+			}
+		}
+
+		class HinhTron extends Hinh 
+		{
+
+			public function chuVi() {
+		 		$n1 = $this->number1;
+				$chuVi = 2 * 3.14 * $n1;
+
+				echo '<div class="ht">Chu vi hình tròn là: ' . $chuVi . '</div><br>';
+			}
+
+			public function dienTich() {
+		 		$n1 = $this->number1;
+				$dienTich = 3.14 * $n1;
+
+				echo '<div class="ht">Diện tích hình tròn là: ' . $dienTich . '</div><br>';		
+			}
+		}
+
+		class TamGiac 
```

Comment:
e ko extends Hinh nên nó mới có lỗi 
`
Notice: Undefined property: TamGiac::$number1 in /opt/lampp/htdocs/pp1906/l6-SC.php on line 102
`
đó e